app/assets/javascripts/controllers/cloud_topology/cloud_topology_controller.js

```diff
@@ -0,0 +1,278 @@
+/* global miqHttpInject */
+
+miqHttpInject(angular.module('cloudTopologyApp', ['kubernetesUI', 'ui.bootstrap', 'ManageIQ']))
+.controller('cloudTopologyController', CloudTopologyCtrl);
+
+CloudTopologyCtrl.$inject = ['$scope', '$http', '$interval', '$location', 'topologyService'];
+
+function CloudTopologyCtrl($scope, $http, $interval, $location, topologyService) {
```

Comment:
I would extend and change the relations like this:

     :tags,
     :availability_zones => [
       :tags,
       :vms => [
         :tags,
         :host,
         :service,
         :genealogy_parent, # image
         :flavor
       ]
     :cloud_tenants      => [
       :tags,
       :children, # cloud tenants can be a tree
       :vms
￼       ],
     :orchestration_stacks => [
        :tags,
        :vms,
        :service,
        :children, # stacks can be a tree
        :cloud_tenant,
        :orchestration_template
￼     ]